bitcoin/core/key.py

```diff
@@ -67,7 +192,10 @@ class CECKey:
     POINT_CONVERSION_UNCOMPRESSED = 4
 
     def __init__(self):
-        self.k = _ssl.EC_KEY_new_by_curve_name(_NID_secp256k1)
+        # WTF: not sure why the cast is necessary here, but we get back an
+        # int without it, even though we set
+        # _ssl.EC_KEY_new_by_curve_name.restype above
+        self.k = ctypes.c_void_p(_ssl.EC_KEY_new_by_curve_name(_NID_secp256k1))
```

Comment:
You are slightly misunderstanding the ctypes system (though note that I am no master). The return type of C functions is always a Python type. Even when you set `restype = c_void_p`, this only means that you get back a (64 bit) `int` or `None`. However, if you set both argument types and return types this way, you *shouldn't* run into conversion issues.

The docs are https://docs.python.org/2/library/ctypes.html#fundamental-data-types, though I don't feel they spell it out clear enough. What comes out is the 'Python type' they mention.